doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+| 9    |   Functionality    | RP is the only receiver (fhr---Interm---rp/lhr)  | |
+| 10   |   Functionality    | Check shared tree on intermediate router for (rp---interm---lhr) scenario  | |
+| 11   |   Functionality    | RP in the middle( fhr---rp---lhr )       | |
+| 12   |   Functionality    | Inverse topology (lhr---fhr---rp)   | |
+| 13   |   Functionality    | Switching from RP-tree to SP-tree   | |
+| 14   |   Functionality    | Verify that router generating an Assert with the lowest Administrative distance is elected as forwarder  | |
+| 15   |   Functionality    | Verify that best unicast routing metric is used to break a tie if the AD are the same  | |
+| 16   |   Functionality    | Verify that device with the highest IP Address will be elected as the PIM Forwarder if the AD and metrics are same  | |
+| 17   |   Functionality    | J/P refresh Interval: random start, 60 default, persistency, join/leave latency  | |
+| 18   |   Functionality    | J/P Holdtime: 3.5* default, oif time out, state time out  | |
+| 19   |   Functionality    | Memory leaking when pim is enabled with multicast traffics| |
+| 20   |   Functionality    | Tree switching and route change  | |  
+| 21   |   Functionality    | Check pim-sm/ssm functionality on non-default vrf context with overlapping address across VRFs for source and / or group addresses  | |
+| 22   |   Functionality    | Jumbo frames validation  | |
+| 23   |   Functionality    | Verify PIM configurations using REST API commands  | |
+| 24   |   Functionality    | Verify PIM show using REST API commands  | |
+| 25   |   Functionality    | Maximum Number of multicast routes/pim group entries supported  | |
+| 26   |   Functionality    | Verify that LHR node stops creating source tree for all the groups when spt-threshold infinity is set globally  | |
+| 27   |   Functionality    | Verify that LHR node stops creating source tree for all the groups mentioned in prefix-list when spt-threshold infinity is set globally with prefix-list  | |
+| 28   |   Functionality    | Verify that no first packet loss is seen with minimal multicast route entries when pim ecmp is configured  | |
+           
+      
+# 11  Design expections from Broadcom
+## 11.1 ACL and COPP requirements
+An ACL to lift unknown multicast address is required to honor the source traffic. The qualifiers and the corresponding actions are listed below.
+
+|    ACL                          | Qualifiers                      |   Action                     |
+|---------------------------------|---------------------------------|------------------------------|
+| Unknown IPv4 Multicast packets must be forwarded to the CPU through a dedicated separate CoPP Queue| PacketRes = Unknown Multicast packets Destination IP=224.0.0.0/4 |   (1) CosQCpuNew -QUEUE_ID_1>  (2) GpCopyToCpu |                      |                                 |                                 |                              |
```

Comment:
Is unknown multicast packets coming on any interface is taken to CPU or only on PIM enabled interfaces. I assume its only on PIM enabled interface, please confirm.

Reply:
Yes, packets on any interface is taken to CPU. We proposed a requirement to add L3 CLASS_IDs but considering the changes in SAI, it was agreed to take up as enhancement. Not covered as part of PIM-SM changes.